sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CsvOutputWriter.scala

```diff
@@ -47,3 +50,15 @@ class CsvOutputWriter(
 
   override def close(): Unit = gen.close()
 }
+
+object CsvOutputWriter {
+
+  def createUnivocityGenerator(
+      maxLength: Int,
+      writer: OutputStreamWriter,
+      dataSchema: StructType,
+      params: CSVOptions): UnivocityGenerator = this.synchronized {
+    NormalizedString.getCache.setMaxStringLength(maxLength)
+    new UnivocityGenerator(dataSchema, writer, params)
```

Comment:
Is the max string length a global setting? If it is, it means `maxColumnNameLength` at different queries can conflict each others?

Reply:
yeah, it's a good question and valid concern...it will

one possible way is to always take the biggest value passed by the user to this config?